pallets/xcm-handler/src/lib.rs

```diff
@@ -598,5 +611,20 @@ pub mod pallet {
 			pending_withdrawals.try_push(pending_withdrawal).unwrap();
 			<PendingWithdrawals<T>>::insert(block_no, pending_withdrawals);
 		}
+
+		pub fn convert_asset_id_to_location(asset_id: u128) -> Option<MultiLocation> {
+			let (_, _, asset_identifier) = <TheaAssets<T>>::get(asset_id);
+			let asset_identifier = asset_identifier.to_vec();
+			let parachain_asset: Option<ParachainAsset> = Decode::decode(&mut &asset_identifier[..]).ok();
+			if let Some(asset) = parachain_asset {
+				Some(asset.location)
+			} else {
+				None
+			}
```

Comment:
I suggest to change this to something like:

```rust
Decode::decode(&mut &asset_identifier[..]).ok().map(|asset: ParachainAsset| asset.location)

```

Reply:
done

Reply:
I assume after resolving conflicts you moved back the old implementation. Please update it again.

Reply:
oops .. I added it back